girder/utility/filesystem_assetstore_adapter.py

```diff
@@ -270,3 +277,61 @@ def cancelUpload(self, upload):
         """
         if os.path.exists(upload['tempFile']):
             os.unlink(upload['tempFile'])
+
+    def importFile(self, item, path, user, name=None, mimeType=None, size=None):
+        """
+        Import a single file from the filesystem into the assetstore.
+
+        :param item: The parent item for the file.
+        :type item: dict
+        :param path: The path on the local filesystem.
+        :type path: str
+        :param user: The user to list as the creator of the file.
+        :type user: dict
+        :param name: Name for the file. Defaults to the basename of ``path``.
+        :type name: str
+        :param mimeType: MIME type of the file if known.
+        :type mimeType: str
+        :param size: Size of the file in bytes. Will stat the file if not
+            passed.
+        :type size: int
+        :returns: The file document that was created.
+        """
+        if size is None:
```

Comment:
As a potential optimization for callers of this method. If they've already done the stat, they can save time by just passing the value in.